source/features/select-all-notifications-shortcut.tsx

```diff
@@ -1,22 +1,34 @@
-import elementReady from 'element-ready';
+import select from 'select-dom';
 import * as pageDetect from 'github-url-detection';
 
 import features from '.';
-import onElementRemoval from '../helpers/on-element-removal';
 
-async function init(signal: AbortSignal): Promise<void> {
-	const selectAllNotifications = await elementReady('.js-notifications-mark-all-prompt');
-	if (selectAllNotifications) { // Notifications page may be empty
-		selectAllNotifications.dataset.hotkey = 'a';
-		await onElementRemoval(selectAllNotifications, signal); // "Select all" checkbox will be replaced if there's more notifications to load #4199
-		void init(signal);
+function runShortcut(event: KeyboardEvent): void {
+	if (
+		event.key === 'a'
+		&& !event.ctrlKey
+		&& !event.metaKey
+		&& !event.isComposing
+	) {
+		select('.js-notifications-mark-all-prompt')!.click();
 	}
 }
 
+function init(signal: AbortSignal): void {
+	// Listen to the shortcut ourselves instead of attaching it to the checkbox #5569
+	document.body.addEventListener('keypress', runShortcut, {signal});
+}
+
 void features.add(import.meta.url, {
+	shortcuts: {
+		a: 'Select all notifications',
+	},
 	include: [
 		pageDetect.isNotifications,
 	],
+	exclude: [
+		() => select.exists('img[src$="notifications/inbox-zero.svg"]'), // Don't run on empty inbox page
```

Comment:
Nit pick
On empty if you click `a` when switching tabs (From participating to Assigned) and its empty

```
Uncaught TypeError: Cannot read properties of undefined (reading 'click')
    at HTMLBodyElement.select_all_notifications_shortcut_runShortcut
```

Reply:
I suppose it would also work if:

1. Select all
2. Mark as done
3. The list is empty
4. Press `a`
5. It throws

But yeah, if that's the procedure, it's not worth fixing

Reply:
> It throws

It doesn't, the "Select all" checkbox stays in the DOM until the page is reloaded

Reply:
I always get it on https://github.com/notifications?query=reason%3Aassign I have nothing there I refreshed a whole bunch of times

Reply:
> I always get it on [github.com/notifications?query=reason%3Aassign](https://github.com/notifications?query=reason%3Aassign) I have nothing there I refreshed a whole bunch of times

Weird, I have no error on Firefox :thinking:

Though I see the feature is running on this tab even when it's empty, because the background image has a slightly different name and thus the `exclude` selector isn't triggered.

We have several notification-related features now, so it's probably worth adding some detections like `isEmptyNotificationPage` and `isNotificationsInboxPage`, that kind of thing.